include/eosio/ship_protocol.hpp

```diff
@@ -15,6 +15,8 @@ template <typename S>
 void to_json(const input_stream& data, S& stream) {
    return to_json_hex(data.pos, data.end - data.pos, stream);
 }
+
+constexpr const char* get_type_name(const input_stream*) { return "bytes";  }
```

Comment:
Should this be moved to stream.hpp ?

Reply:
Probably, but there's a dependency issue: `to_json.hpp` depends on `stream.hpp`, so we can't put anything that depends on to_json in `stream.hpp` or the dependencies become circular.